mlrun/datastore/store_resources.py

```diff
@@ -0,0 +1,128 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# flake8: noqa  - this is until we take care of the F401 violations with respect to __all__ & sphinx
+
+import mlrun
+from mlrun.config import config
+from mlrun.utils.helpers import parse_function_uri
+from .v3io import v3io_path
+from ..utils import StorePrefix, parse_store_uri
+
+
+class ResourceCache:
+    """Resource cache for real-time pipeline/serving and storey"""
+
+    def __init__(self):
+        self._tabels = {}
+        self._resources = {}
+
+    def cache_table(self, uri, value, is_default=False):
+        """Cache storey Table objects"""
+        self._tabels[uri] = value
+        if is_default:
+            self._tabels["."] = value
+
+    def get_table(self, uri):
+        """get storey Table object by uri"""
+        try:
+            from storey import Table, Driver, V3ioDriver
+        except ImportError:
+            raise ImportError("storey package is not installed, use pip install storey")
+
+        if uri in self._tabels:
+            return self._tabels[uri]
+        if uri in [".", ""]:
+            self._tabels[uri] = Table("", Driver())
+            return self._tabels[uri]
+
+        if uri.startswith("v3io://") or uri.startswith("v3ios://"):
+            endpoint, uri = v3io_path(uri)
+            self._tabels[uri] = Table(uri, V3ioDriver(webapi=endpoint))
+            return self._tabels[uri]
+
+        # todo: map store:// uri's to Table objects
+
+        raise ValueError(f"table {uri} not found in cache")
+
+    def cache_resource(self, uri, value, default=False):
+        """cache store resource (artifact/feature-set/feature-vector)"""
+        self._resources[uri] = value
+        if default:
+            self._resources["."] = value
+
+    def get_resource(self, uri):
+        """get resource from cache by uri"""
+        return self._resources[uri]
+
+    def resource_getter(self, db=None, secrets=None):
+        """wraps get_store_resource with a simple object cache"""
+
+        def _get_store_resource(uri, use_cache=True):
+            if (uri == "." or use_cache) and uri in self._resources:
+                return self._resources[uri]
+            resource = get_store_resource(uri, db, secrets=secrets)
+            if use_cache:
+                self._resources[uri] = resource
+            return resource
+
+        return _get_store_resource
+
+
+def get_store_resource(uri, db=None, secrets=None, project=None):
+    """get store resource object by uri"""
+
+    db = db or mlrun.get_run_db(secrets=secrets)
+    kind, uri = parse_store_uri(uri)
+    if kind == StorePrefix.FeatureSet:
+        project, name, tag, uid = parse_function_uri(
```

Comment:
I guess you didn't want to rename `parse_function_uri` cause you didn't want to go and change all of the places that are really function related that are calling it
therefore I suggest that you'll add a function `parse_versioned_object_uri` which you'll use everywhere in your code, and this function will inside simply call `parse_function_uri` (with an explanation why we don't merge the 2 functions)

Reply:
replaced all w `parse_versioned_object_uri`, not in too many places